webapp/src/utils/index.ts

```diff
@@ -0,0 +1,28 @@
+import {id} from 'manifest';
+
+const getBaseUrls = (): {pluginApiBaseUrl: string; mattermostApiBaseUrl: string} => {
+    const url = new URL(window.location.href);
+    const baseUrl = `${url.protocol}//${url.host}`;
+    const pluginUrl = `${baseUrl}/plugins/${id}`;
+    const pluginApiBaseUrl = `${pluginUrl}/api/v1`;
+    const mattermostApiBaseUrl = `${baseUrl}/api/v4`;
+
+    return {pluginApiBaseUrl, mattermostApiBaseUrl};
+};
```

Comment:
don't use `window` it can be vulnerable to attacks, you can see https://github.com/Brightscout/mattermost-plugin-template/blob/bfef1df581b40461702d166d7007f40509cb76a3/webapp/src/utils/index.ts#L6